### PR TITLE
Add query execution stats for 1.1.0

### PR DESCRIPTION
**Pull request by simonz-bq**
- Added query execution stats with unit and integration tests
- Bumped version to 1.1.0
- Updated dependencies
- Removed year on Copyright

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Amazon.QLDB.Driver.Tests/ResultTests.cs [hunk 1]
```diff
@@ -108,5 +151,201 @@ public void TestIonEnumeratorResetIsNotSupported()
...
+            timing = result.GetTimingInformation();
+            Assert.AreEqual(fetchReads, io.ReadIOs);
+            Assert.AreEqual(fetchWrites, io.WriteIOs);
+            Assert.AreEqual(fetchTime, timing.ProcessingTimeMilliseconds);
+        }
+
+        [TestMethod]
+        public void TestQueryStatsHasExecuteNullFetch()
+        {
+            Mock<Session> mockSession = new Mock<Session>(null, null, null, null, null);
```
Comment (alpian) [inline Amazon.QLDB.Driver.Tests/ResultTests.cs:254]: There seems to be a lot of duplication between tests. Can we clean that up please?

File: Amazon.QLDB.Driver.Tests/ResultTests.cs [hunk 2]
```diff
@@ -108,5 +151,201 @@ public void TestIonEnumeratorResetIsNotSupported()
...
+            timing = result.GetTimingInformation();
+            Assert.AreEqual(executeReads, io.ReadIOs);
+            Assert.AreEqual(executeWrites, io.WriteIOs);
+            Assert.AreEqual(executeTime, timing.ProcessingTimeMilliseconds);
+        }
+
+        [TestMethod]
+        public void TestQueryStatsHasExecuteHasFetch()
+        {
+            Mock<Session> mockSession = new Mock<Session>(null, null, null, null, null);
```
Comment (alpian) [inline Amazon.QLDB.Driver.Tests/ResultTests.cs:303]: Duplicated from above

File: Amazon.QLDB.Driver/client/IOUsage.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (alpian) [inline Amazon.QLDB.Driver/client/IOUsage.cs:2]: 2021

File: Amazon.QLDB.Driver/client/TimingInformation.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (alpian) [inline Amazon.QLDB.Driver/client/TimingInformation.cs:2]: 2021

File: Amazon.QLDB.Driver/result/BufferedResult.cs [hunk 5]
```diff
@@ -25,15 +25,21 @@ namespace Amazon.QLDB.Driver
     public class BufferedResult : IResult
     {
         private readonly List<IIonValue> values;
+        private readonly IOUsage consumedIOs;
+        private readonly TimingInformation timingInformation;
```
Comment (alpian) [inline Amazon.QLDB.Driver/result/BufferedResult.cs:29]: Are these nullable?

File: Amazon.QLDB.Driver/result/IResult.cs [hunk 6]
```diff
@@ -22,5 +22,18 @@ namespace Amazon.QLDB.Driver
     /// </summary>
     public interface IResult : IEnumerable<IIonValue>
     {
+        /// <summary>
+        /// Gets the current query statistics for the number of read and write IO requests. The statistics are stateful.
+        /// </summary>
+        ///
+        /// <returns>The current IOUsage statistics.</returns>
+        IOUsage GetConsumedIOs();
```
Comment (alpian) [inline Amazon.QLDB.Driver/result/IResult.cs:30]: Are these Nullable?

**Review by alpian — CHANGES_REQUESTED** [review]

File: Amazon.QLDB.Driver/result/BufferedResult.cs (hunk 5 above)
Comment (simonz-bq) [reply Amazon.QLDB.Driver/result/BufferedResult.cs:29]: No, but the contained int64s are.

File: Amazon.QLDB.Driver/result/IResult.cs (hunk 6 above)
Comment (simonz-bq) [reply Amazon.QLDB.Driver/result/IResult.cs:30]: No, but the contained int64s are.

File: Amazon.QLDB.Driver/client/TimingInformation.cs (hunk 4 above)
Comment (simonz-bq) [reply Amazon.QLDB.Driver/client/TimingInformation.cs:2]: Done

File: Amazon.QLDB.Driver/client/IOUsage.cs (hunk 3 above)
Comment (simonz-bq) [reply Amazon.QLDB.Driver/client/IOUsage.cs:2]: Done

**Review by alpian — CHANGES_REQUESTED** [review]
Please address the other comments not added as part of the review (sorry, my bad for not knowing how GH works :))

**Review by alpian — APPROVED** [review]
Nice! :)

**Pull request merged**